async-clipboard/demo.js

```diff
@@ -0,0 +1,55 @@
+let copy = document.querySelector('#copy');
+let paste = document.querySelector('#paste');
+let out = document.querySelector('#out');
+
+/** Write contents of the textarea to the clipboard when clicking "Copy" */
+copy.onclick = () => {
+  navigator.clipboard.writeText(out.value)
+    .then(result => {
+      ChromeSamples.log('Text copied.' + result);
+    })
+    .catch(err => {
+      ChromeSamples.log('Failed to copy text: ' + err);
+    });
+};
+
+/** Read from clipboard when clicking the Paste button */
+paste.onclick = () => {
+  navigator.clipboard.readText()
+    .then(text => {
+      out.value = text;
+      ChromeSamples.log('Text pasted.');
+    })
+    .catch(err => {
+      ChromeSamples.log('Failed to read from clipboard: ' + err);
+    });
+};
+
+/** Watch for pastes */
+document.addEventListener('paste', e => {
+  e.preventDefault();
+  navigator.clipboard.getText().then(text => {
+    ChromeSamples.log('Updated clipboard contents: ' + text);
+  });
+});
+
+/** Set up buttons to request permissions and display status: */
+[].forEach.call(document.querySelectorAll('data-permission'), btn => {
```

Comment:
Since this api will only work in browsers that support `NodeList.forEach`, might as well use that :)